### DISCUSSION TITLE
A Rust book for RayTracing in One Week?

### DISCUSSION THREAD

**Discussion by CaioLC** [RayTracing/raytracing.github.io#1237] [Ideas]
Hi guys,

I've been working on a Rust implementation of the RT in One Week book on my spare time as some coding practice. It's been very entertaining and a great opportunity to explore some rust's mid-level concepts such as lifetimes and trait objects. 

I had the idea of sharing this experience by writing a proper rust book, following the steps of RayTracing in One Week. The idea is _not_ to rewrite the book, but instead write some sort of a journal with the corresponding rust code and a few additions of mine to discuss rust specific concepts. Of course, it would also be public domain (book and code) and with due credits to the original book and authors. Not sure what are your opinions on these kinds of derivative works so here's me reaching out :).

**Comment by hollasch** [comment]
Sounds good to me!

**Reply by CaioLC** [reply]
Great. I'll share a link here once I start writing. Any tips / feedback will be much appreciated
